src/logging.rs

```diff
@@ -97,6 +105,47 @@ pub mod trace {
         }
     }
 
+    impl LevelHandle {
+        /// Returns a new `LevelHandle` without a corresponding filter.
+        ///
+        /// This will do nothing, but is required for admin endpoint tests which
+        /// do not exercise the `proxy-log-level` endpoint.
+        pub fn dangling() -> Self {
+            let builder = subscriber_builder()
+                .with_filter(filter::EnvFilter::default())
+                .with_filter_reloading();
+            let inner = builder.reload_handle();
+            LevelHandle { inner }
+        }
+
+        pub fn set_level(&self, level: impl AsRef<str>) -> Result<(), Error> {
```

Comment:
It's surprising that this isn't `&mut self`?

Reply:
The `reload` method on `tracing-fmt::filter::reload::Handle` only requires `&self` because the actual filter state is behind a read-write lock; setting a new filter acquires the write lock. So, there's no _reason_ for this to take a mutable receiver.